src/extension.ts

```diff
@@ -407,22 +407,25 @@ class DeviceTreeItem extends vscode.TreeItem {
         });
     }
 
-    public async printSysinfo() {
-        const messageBarItem = new StatusBarProgressionMessage("Grabbing ev3dev system info...");
-        try {
-            const sysinfo = await this.device.getSystemInfo();
-
-            output.appendLine("========== ev3dev-sysinfo ==========");
-            output.appendLine(sysinfo);
-            output.appendLine("");
-            output.show();
-
-            messageBarItem.finish("System info downloaded; see output window.");
-        }
-        catch(err) {
-            vscode.window.showErrorMessage("An error occurred while capturing system info: " + (err.message || err));
-            messageBarItem.finish();
-        }
+    public async showSysinfo() {

```

Comment:
Explicit return type please.

    public async showSysinfo(): Promise<void> {

Reply:
I'm going to push back on that... why? For async functions that return values this would make sense, but in this case the convention that I've used (and the one that I've seen used in most samples) is to omit the return type when no value is returned -- async or otherwise.

Reply:
The fact that you have this declaration actually shows that there is a bug here. Typescript requires that you return a Promise for async functions, but there is no await here, so async is not needed.

Reply:
I'm... not sure what you mean. The call is `await`ed. https://github.com/WasabiFan/vscode-ev3dev-browser/blob/cc3e411e62744bd4a66f1b6ce3ebf956f05f8602/src/extension.ts#L412

Reply:
Hmm... I see there is an await here. The typescript compiler has not let me get away with having async without a return type so far.